### PR TITLE
Add fact which reports xcode version

### PR DESCRIPTION
**Pull request by seanabraham**
I still need to test this, but putting it up pre-emptively

### CODE REVIEW

File: lib/facter/xcode_version.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'puppet/util/plist' if Puppet.features.cfpropertylist?
+
+xcode_version_plist = '/Applications/Xcode.app/Contents/version.plist'
```
Comment (seanabraham) [inline lib/facter/xcode_version.rb]: I would have just used the `xcode_active_directory` here but if that's `CommandLineTools` for example we won't have `version.plist`

**Comment by seanabraham** [comment]
Tested this on my machine `$ FACTERLIB=$(pwd):$FACTERLIB facter -p xcode_version` and it worked as expected

**Comment by seanabraham** [comment]
cc @grahamgilbert

**Comment by grahamgilbert** [comment]
I wonder if we could use `xcode_active_directory` if it contains .app and then fall back the default location if that is just the command line tools. That means we get the version being used if there is more than one Xcode present.

**Comment by seanabraham** [comment]
@grahamgilbert I can give that a whirl, just check for `.app` in the actual path? One other kinda annoying part of that though is that `xcode_active_directory` returns `/Applications/Xcode.app/Contents/Developer`, so I'd have to go up one directory to `Contents` to find the `verison.plist`. Will that always be the case with `xcode_active_directory` (provided that it doesn't return command line tools)

**Pull request merged**